File: src/grader/ex00e4/gen_barcode.rb
def gen(n, sol, len, sum, k)
  if (len < n)
    if (sum <= k)
      sol += "0"
      gen(n, sol, len + 1, sum, k)
      sol[len] = "1"
      gen(n, sol, len + 1, sum + 1, k)
    end
    return
  else
    if (sum == k)
      puts sol
    end
    return
  end
end

k, n = gets.split.map { |x| x.to_i }
gen(n, "", 0, 0, k)

File: src/grader/dc&dp/ex00e4/gen_barcode.cpp
#include <stdio.h>

void gen(int n, int* sol, int len, int sum, int k) {
    if (len < n) {
        sol[len] = 0;
        gen(n, sol, len + 1, sum, k);
        sol[len] = 1;
        gen(n, sol, len + 1, sum + 1, k);
    } else {
        if (sum == k) {
            for (int i = 0; i < n; ++i) {
                printf("%d", sol[i]);
            }
            printf("\n");
        }
        return;
    }
}

int main() {
    int n, k;
    scanf("%d%d", &k, &n);
    int sol[n];
    gen(n, sol, 0, 0, k);
}